bloxid/hashids_test.go

```diff
@@ -0,0 +1,184 @@
+package bloxid
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHashIDtoInt(t *testing.T) {
+	var tests = []struct {
+		name   string
+		hashID string
+		intID  int
+		salt   string
+		err    error
+	}{
+		{
+			name:   "Valid input",
+			hashID: "blox0.hostapp.us-com-1.jbeuiwrsmq3tkmzwmuzwcojsmrqwemrtgy3tqzbvhbsdizjvhe2dkn3cgzrdizlb",
+			intID:  1,
+			salt:   "test",
+			err:    nil,
+		},
+		{
+			name:   "Different salt",
+			hashID: "blox0.hostapp.us-com-1.jbeuiwrsmq3tkmzwmuzwcojsmrqwemrtgy3tqzbvhbsdizjvhe2dkn3cgzrdizlb",
+			intID:  -1,
+			salt:   "testi1",
+			err:    errors.New("mismatch between encode and decode: 2d7536e3a92dab23678d58d4e59457b6b4ea start ed4b2a9764524ed6958d58237ba3eadb7695 re-encoded. result: [4]"),
+		},
+
+		{
+			name:   "Invalid prefix HIDA with correct int value",
+			hashID: "blox0.hostapp.us-com-1.jbeuiqjsmq3tkmzwmuzwcojsmrqwemrtgy3tqzbvhbsdizjvhe2dkn3cgzrdizlb",
+			intID:  -1,
+			salt:   "test",
+			err:    nil,
+		},
+	}
+
+	for _, test := range tests {
+
+		t.Run(test.name, func(t *testing.T) {
+			v0_1, err := NewV0(test.hashID, WithDecodeHashIDSalt(test.salt))
+
+			assert.Equal(t, test.err, err)
+			assert.Equal(t, test.intID, v0_1.HashID())
+		})
+	}
+}
+
+func TestHashIDInttoInt(t *testing.T) {
+	var tests = []struct {
+		name       string
+		intID      int
+		entityType string
+		region     string
+		salt       string
+		err        error
+	}{
+		{
+			name:       "Valid input",
+			intID:      1,
+			entityType: "hostapp",
+			region:     "us-com-1",
+			salt:       "test",
+			err:        nil,
+		},
+		{
+			name:       "Negative number - becomes entropy",
+			intID:      -1,
+			entityType: "hostapp",
+			region:     "us-com-1",
+			salt:       "test",
+			err:        nil,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			v0, err := GenerateV0(&V0Options{
```

Comment:
please change to this sig 
```suggestion
			v0, err := NewV0("", WithEncodeOptions(
    WithEntityType(test.entityType),
    WithRegion(test.region),
    WithHashID(test.intID),
    WithHashIDSalt(test.salt)
)

```